src/cls/journal/cls_journal.cc

```diff
@@ -540,10 +540,12 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
                             bufferlist *out) {
   std::string id;
   bufferlist data;
+  std::set<cls::journal::Client> clients;
   try {
     bufferlist::iterator iter = in->begin();
     ::decode(id, iter);
     ::decode(data, iter);
+    ::decode(clients, iter);
```

Comment:
Calling journal_client_list() was my initial approach, but that meant having similar aggregation logic as in C_ClientList::complete(). Maybe, treat `::decode(clients, iter)` specially in another `try..catch` block and handle failures (falling back to older behaviour)? However, that skips initializing commit positions.